src/app/learnocaml_partition_view.ml

```diff
@@ -80,7 +89,36 @@ let render_classes xs =
 
 let sum_with f = List.fold_left (fun acc x -> acc + f x) 0
 
-let exercises_tab part =
+let rec students_partition students part =
+  let open Student in
+  match part with
+    |[] -> []
+    |t::q -> let student_t = List.find (fun student -> student.token = t) students in
+             student_t :: students_partition students q
```

Comment:
Traversing `part`, the token list once? fine.
But doing one traversal of (potentially) the whole list of `students` for any token looks like suboptimal.
I'd suggest turning first the `students` list into a hash Map.
In this case, you can do:

```suggestion
let students_partition (students : Student.t Token.Map.t) part =
  List.map (fun t -> Token.Map.find t students) part
```